ietf/sync/rfceditor.py

```diff
@@ -367,80 +370,140 @@ def update_docs_from_rfc_index(index_data, errata_data, skip_older_than_date=Non
+    tag_has_errata = DocTagName.objects.get(slug="errata")
+    tag_has_verified_errata = DocTagName.objects.get(slug="verified-errata")
     relationship_obsoletes = DocRelationshipName.objects.get(slug="obs")
     relationship_updates = DocRelationshipName.objects.get(slug="updates")
 
     system = Person.objects.get(name="(System)")
 
-    for rfc_number, title, authors, rfc_published_date, current_status, updates, updated_by, obsoletes, obsoleted_by, also, draft, has_errata, stream, wg, file_formats, pages, abstract in index_data:
-
+    for (
+        rfc_number,
+        title,
+        authors,
+        rfc_published_date,
+        current_status,
+        updates,
+        updated_by,
+        obsoletes,
+        obsoleted_by,
+        also,
+        draft_name,
+        has_errata,
+        stream,
+        wg,
+        file_formats,
+        pages,
+        abstract,
+    ) in index_data:
         if skip_older_than_date and rfc_published_date < skip_older_than_date:
             # speed up the process by skipping old entries
             continue
 
         # we assume two things can happen: we get a new RFC, or an
         # attribute has been updated at the RFC Editor (RFC Editor
         # attributes take precedence over our local attributes)
-        events = []
-        changes = []
+        rfc_events = []
+        rfc_changes = []
         rfc_published = False
+        draft_events = []
+        draft_changes = []
 
-        # make sure we got the document and alias
-        doc = None
-        name = "rfc%s" % rfc_number
-        a = DocAlias.objects.filter(name=name)
-        if a:
-            doc = a[0].document
-        else:
-            if draft:
-                try:
-                    doc = Document.objects.get(name=draft)
-                except Document.DoesNotExist:
-                    pass
-
-            if not doc:
-                changes.append("created document %s" % prettify_std_name(name))
-                doc = Document.objects.create(name=name, type=DocTypeName.objects.get(slug="draft"))
-
-            # add alias
-            alias, __ = DocAlias.objects.get_or_create(name=name)
+        # Find the document
+        doc, created_rfc = Document.objects.get_or_create(
+            rfc_number=rfc_number, type_id="rfc", defaults={"name": f"rfc{rfc_number}"}
+        )
+        if created_rfc:
+            rfc_changes.append(f"created document {prettify_std_name(doc.name)}")
+            # Create DocAlias (for consistency until we drop DocAlias altogether)
+            alias, _ = DocAlias.objects.get_or_create(name=doc.name)
             alias.docs.add(doc)
-            changes.append("created alias %s" % prettify_std_name(name))
+            rfc_changes.append(f"created alias {prettify_std_name(doc.name)}")
+
+        if draft_name:
+            try:
+                draft = Document.objects.get(name=draft_name)
+            except Document.DoesNotExist:
+                pass
```

Comment:
We should probably log this - the RPC thinks the doc came from a draft, but we can't find it. That's probably a real error.

Reply:
will do